### PR TITLE
Improved support for Enterprise Linux family OSs (CentOS, RHEL etc)

### PR DESCRIPTION
**Pull request by mattwillsher**
* Install pygit from package
* Add Official Salt ZeroMQ 4 COPR repository

Note that Salt itself is assumed to already to be available to the system via yum, via EPEL for example

### CODE REVIEW

File: salt/pkgrepo/init.sls [hunk 1]
```diff
@@ -1,4 +1,6 @@
-{% if grains['os_family'] == 'Debian' %}
+{% set name = {
+    'RedHat': 'redhat',
+    'Debian': grains['os']|lower,
+}.get(grains.os_family) %}
 include:
-  - .{{ grains['os']|lower }}
-{% endif %}
+  - .{{ name }}
```
Comment (iggy) [inline salt/pkgrepo/init.sls]: You probably could have just changed `grains['os']|lower` to `grains['os_family']|lower`

Comment (iggy) [inline salt/pkgrepo/init.sls]: Nevermind, I didn't realize there was already stuff for not just Debian there.

**Comment by aboe76** [comment]
@mattwillsher did the master config part in map.jinja work on rhel/centos?
because I don't see f_defaults.conf change with pygit2 instead of gitpython...

**Comment by mattwillsher** [comment]
Now you mention it, no I think it got left unset. I can take a look tomorrow. From memory, it being unset meant that pygit2 got used as it was available.

What is it it you see? That when gitpython is set it gets left as it is rather that replaced by pygit?

_____________________________

**Comment by aboe76** [comment]
You have set the map.jinja redhat default master: gitfs_provider: pygit2,

What I would think that should happen is that the f_defaults.conf picks this up and uses the parameter.
From what I can see is that all master settings in map.jinja aren't used in the template files.
only in the state part.

**Comment by mattwillsher** [comment]
pygit2 is the default in any case. map.jinja is currently only used by the formula itself as you point out. While I agree the value should be set  in the f_defaults, this has never been the case for the formula, if I understand correctly. Perhaps this is better opened as an issue and tracked via that?

**Pull request merged**